src/Featurizers/Components/WindowFeaturizerBase.h

```diff
@@ -293,6 +295,11 @@ CircularBuffer<T>::CircularBuffer(size_t max_size) : _max_size(max_size), _start
     _data.reserve(_max_size);
 }
 
+template <class T>
+size_t CircularBuffer<T>::size() const {
+    return _data.size();
+}
+
```

Comment:
I added this method too, so if my PR goes in first you should revert this file, and if yours goes in first ill revert mine. Easier to avoid a conflict that way.

Reply:
Yep. I just added it so it could compile for now. I guess this method and change in GrainEstimator constructor can be both included in a separate PR.